presto-main-base/src/main/java/com/facebook/presto/sql/planner/iterative/Memo.java

```diff
@@ -110,6 +110,38 @@ public PlanNode getNode(int group)
         return getGroup(group).membership;
     }
 
+    public int replicateGroup(int oldGroupId, PlanNode newReference)
```

Comment:
what's the benefit of doing this vs. inserting the new reference to a new group (via insertRecursive())?

 This method looks similar to replace() where you are trying to keep the group, but replace it's membership, so it needs to specifically only get the parts of the group that it wants to change.  Here since we want to basically insert a copy of the original node again, inserting it should be simpler (and safer, since there is nothing that guarantees that the caller passes a newReference that is a copy of the old reference).